Bring PR information inline with confluence (#208)
netsells/code-standards: src/general/pull-requests.md

## Overview
This document details what is expected of you and what you can expect during the PR review process. This guide is not definitive or exhaustive but it does cover many of the common obstacles that we all come across. Remember that the intention of a review is to ensure the code quality is as high as possible and discussion is always welcome.

> We run all of our code through pull requests for a variety of reasons, these include:

Reviewer: ```suggestion
We run all of our code through pull requests for a variety of reasons. These include:
```

To request a review, select the people that you want from the dropdown panel. You should add a reviewer for each discipline that your PR changes, for example if you build a tool for Nova admin panel you will need both a frontend and backend reviewer. For large or complicated changes, several reviewers are recommended to ensure multiple sets of eyes see them. You should aim to bring in a more senior developer to review your code and junior developers can be added if you believe reviewing the code would be of benefit to them.

## Review Comments

> When leaving a review, we use two types of pull request comments, **MUST** and **SHOULD**, these keywords will indicate the requirement level of a comment. If a reviewer leaves a **MUST** comment, the pull request will be rejected pending changes. It is up to the reviewers discretion to decide whether a PR is rejected due to **SHOULD** comments. Comments can also be left without this syntax which could be a general question asking you for more information or a piece of information that you may not know. 

Reviewer 2: Suggestions aren't quite as nice in a .md file! We have a lot of commas in this sentence- maybe splitting it into two would read better?
```suggestion
When leaving a review, we use two types of pull request comments; **MUST** and **SHOULD**. These keywords will indicate the requirement level of a comment. If a reviewer leaves a **MUST** comment, the pull request will be rejected pending changes. It is up to the reviewers discretion to decide whether a PR is rejected due to **SHOULD** comments. Comments can also be left without this syntax which could be a general question asking you for more information or a piece of information that you may not know. 
```

Comments should be critical, informative and descriptive enough for you to implement the suggestion. Comments such as “this is wrong” or “this doesn’t work” are not acceptable and should be replaced with an explanation of the problem and suggestions for the solution, including code snippets where appropriate.

Examples of appropriate comments;

`MUST: It looks like you've missed an import for the DB facade, make sure you prefix with a \ to avoid exceptions!`

```
SHOULD: we typically split arrays onto multiple lines for better gif diffs (with a trailing comma)

'user' => [
    'first_name', 
    'last_name',
],
```

## Comment Tips and Examples for Reviewers
- Try to give clear instructions on what you want changing and do not refer to in-person discussions. Pull requests are often looked back on to understand why decisions were made, "as we spoke about on Tuesday" does not help anyone 5 months later!
- If an in-person discussion is needed (ie it'll be a lengthy comment thread) - make sure you post back with the summary so it's known to others

> - Provide clear examples whenever possible to help mentor and avoid too many back and forth with clarifications.

Reviewer 2: ```suggestion
- Provide clear examples whenever possible to help mentor and avoid excessive back and forth with clarifications.
```

- Read every line of code
- Don't hold author hostage - We have enough power to block a PR, lets not use this for evil by requesting changes outside the scope of the PR. e.g fixing old code

> 

Reviewer 2: ```suggestion
- Add links to comments where appropriate. External docs and discussions around topics are extremely valuable learning resources.

```

To request a review, select the people that you want from the dropdown panel. You should add a reviewer for each discipline that your PR changes, for example if you build a tool for Nova admin panel you will need both a frontend and backend reviewer. For large or complicated changes, several reviewers are recommended to ensure multiple sets of eyes see them. You should aim to bring in a more senior developer to review your code and junior developers can be added if you believe reviewing the code would be of benefit to them.

## Review Comments

> When leaving a review, we use two types of pull request comments; **MUST** and **SHOULD**. These keywords will indicate the requirement level of a comment. If a reviewer leaves a **MUST** comment, the pull request will be rejected pending changes. It is up to the reviewers discretion to decide whether a PR is rejected due to **SHOULD** comments. Comments can also be left without this syntax which could be a general question asking you for more information or a piece of information that you may not know. 

Reviewer: ```suggestion
When leaving a review, we use two types of pull request comments: **MUST** and **SHOULD**. These keywords will indicate the requirement level of a comment. If a reviewer leaves a **MUST** comment, the pull request will be rejected pending changes. It is up to the reviewers discretion to decide whether a PR is rejected due to **SHOULD** comments. Comments can also be left without this syntax which could be a general question asking you for more information or a piece of information that you may not know. 
```

Comments should be critical, informative and descriptive enough for you to implement the suggestion. Comments such as “this is wrong” or “this doesn’t work” are not acceptable and should be replaced with an explanation of the problem and suggestions for the solution, including code snippets where appropriate.

Examples of appropriate comments;

`MUST: It looks like you've missed an import for the DB facade, make sure you prefix with a \ to avoid exceptions!`

```
SHOULD: we typically split arrays onto multiple lines for better gif diffs (with a trailing comma)

'user' => [
    'first_name', 
    'last_name',
],
```

## Comment Tips and Examples for Reviewers

> - Try to give clear instructions on what you want changing and do not refer to in-person discussions. Pull requests are often looked back on to understand why decisions were made, "as we spoke about on Tuesday" does not help anyone 5 months later!

Reviewer: ```suggestion
- Try to give clear instructions on what you want changing and do not refer to in-person discussions. Pull requests are often looked back on to understand why decisions were made; "as we spoke about on Tuesday" does not help anyone 5 months later!
```

To request a review, select the people that you want from the dropdown panel. You should add a reviewer for each discipline that your PR changes, for example if you build a tool for Nova admin panel you will need both a frontend and backend reviewer. For large or complicated changes, several reviewers are recommended to ensure multiple sets of eyes see them. You should aim to bring in a more senior developer to review your code and junior developers can be added if you believe reviewing the code would be of benefit to them.

## Review Comments

> When leaving a review, we have two types of pull request comments, **MUST** and **SHOULD**, these keywords to indicate the level of a comment. If a reviewer leaves a **MUST** comment, the pull request will be rejected pending changes. It is up to the reviewers discretion to decide whether a PR is rejected due to **SHOULD** comments. Comments can also be left without this syntax which could be a general question asking you for more information or a piece of information that you may not know. 

Reviewer 3: First sentence doesn't quite read right

Comments should be critical, informative and descriptive enough for you to implement the suggestion. Comments such as “this is wrong” or “this doesn’t work” are not acceptable and should be replaced with an explanation of the problem and suggestions for the solution, including code snippets where appropriate.

> 

Reviewer 3: Lots of white space here
